lib/checknullpointer.cpp

```diff
@@ -509,31 +509,49 @@ void CheckNullPointer::arithmetic()
     for (std::size_t i = 0; i < functions; ++i) {
         const Scope * scope = symbolDatabase->functionScopes[i];
         for (const Token* tok = scope->classStart->next(); tok != scope->classEnd; tok = tok->next()) {
-            if (!tok->astOperand2() || tok->str() != "-")
-                continue;
-            // pointer subtraction
-            if (!tok->valueType() || !tok->valueType()->pointer)
-                continue;
-            // Can LHS be NULL?
-            const ValueFlow::Value *value = tok->astOperand1()->getValue(0);
-            if (!value)
-                continue;
-            if (!_settings->inconclusive && value->isInconclusive())
-                continue;
-            if (value->condition && !_settings->isEnabled(Settings::WARNING))
-                continue;
-            arithmeticError(tok,value);
+            if (tok->str() == "-" || tok->str() == "+" || tok->str() == "+=" || tok->str() == "-=")
```

Comment:
I prefer early continue. And a `Token::Match` should be used:

    if (!Token::Match(tok, "-|+|+=|-="))
        continue;